### PR TITLE
OHM-47: Updated javascript generators to coffeescript

### PR DESCRIPTION
**Pull request by BMartinos**
This PR relates to issue #343. 
Some additional changes to the other generators as well. ( code consistency )

### CODE REVIEW

**Comment by BMartinos** [comment]
Hey @devcritter Can you review this PR for me when you get a chance. Thanks

File: src/api/contactGroups.coffee [hunk 1]
```diff
@@ -3,148 +3,110 @@ Q = require 'q'
...
 
+utils = require "../utils"
 
 ###############################
 #     Adds a contactGroup     #
 ###############################
-exports.addContactGroup = `function *addContactGroup() {
+exports.addContactGroup = ->
+  # Must be admin
+  if authorisation.inGroup('admin', @authenticated) is false
```
Comment (devcritter) [inline src/api/contactGroups.coffee]: not really a big deal, but it's a bit nicer to say:
```
if not authorisation.inGroup 'admin', this.authenticated
```

File: src/api/contactGroups.coffee [hunk 2]
```diff
@@ -3,148 +3,110 @@ Q = require 'q'
...
+    return
 
-  // Test if the user is authorised
-  if (authorisation.inGroup('admin', this.authenticated) === false) {
-    logger.info('User ' +this.authenticated.email+ ' is not an admin, API access to addContactGroup denied.')
-    this.body = 'User ' +this.authenticated.email+ ' is not an admin, API access to addContactGroup denied.'
-    this.status = 'forbidden';
-    return;
-  }
+  contactGroupData = @request.body
```
Comment (devcritter) [inline src/api/contactGroups.coffee]: as discussed in hipchat, let's stick to `this.`

File: src/api/contactGroups.coffee [hunk 3]
```diff
@@ -3,148 +3,110 @@ Q = require 'q'
...
-
-  }
-}`
+exports.getContactGroup = (contactGroupId) ->
+  # Must be admin
+  if authorisation.inGroup('admin', @authenticated) is false
+    utils.logAndSetResponse this, 'forbidden', "User #{this.authenticated.email} is not an admin, API access to getContactGroup denied.", 'info'
+    return
+
+  contactGroupId = unescape(contactGroupId)
```
Comment (devcritter) [inline src/api/contactGroups.coffee]: here and in a few other places: you can remove the (): `unescape contactGroupId`

File: src/api/visualizer.coffee [hunk 4]
```diff
@@ -4,12 +4,12 @@ exports.getLatestEvents = ->
...
     result = yield VisualizerEvent.find({ 'created': { '$gte': rtDate } }).sort({ 'ts': 1 }).exec()
-    this.body = { events: result }
+    @body = { events: result }
   catch err
     logAndSetResponse this, 'internal server error', "Could not fetch the latest visualizer events via the API: #{err}", 'error'
 
 exports.sync = ->
   try
-    this.body = { now: Date.now() }
+    @body = { now: Date.now() }
```
Comment (devcritter) [inline src/api/visualizer.coffee]: The braces aren't really necessary anymore: `this.body = now: Date.now()` should work fine. But not a big deal

**Comment by devcritter** [comment]
Thanks @BMartinos just some minor comments

**Comment by BMartinos** [comment]
Hey @devcritter I have made the corrections as described in the above comment. Can you review again please. Thanks.

**Comment by devcritter** [comment]
I think you missed api/mediators.coffee?

**Comment by BMartinos** [comment]
yip sorry about that @devcritter :). Ive uploaded the mediator.coffee as well

**Comment by devcritter** [comment]
Cool :) Looks good :+1:

**Pull request merged**